packages/@aws-cdk/aws-ec2/lib/cfn-init-elements.ts

```diff
@@ -0,0 +1,405 @@
+   *
+   * @default - Do not restart any service
+   */
+  readonly serviceRestartHandles?: InitServiceRestartHandle[];
+}
+
+/**
+ * Command to execute on the instance
+ */
+export class InitCommand extends InitElement {
+  /**
+   * Run a shell command
+   *
+   * You must escape the string appropriately.
+   */
+  public static shellCommand(shellCommand: string, options: InitCommandOptions = {}): InitCommand {
+    return new InitCommand(shellCommand, options);
+  }
+
+  /**
+   * Run a command from an argv array
+   *
+   * You do not need to escape space characters or enclose command parameters in quotes.
+   */
+  public static argvCommand(argv: string[], options: InitCommandOptions = {}): InitCommand {
+    if (argv.length === 0) {
+      throw new Error('Cannot define argvCommand with an empty arguments');
+    }
+    return new InitCommand(argv, options);
+  }
+
+  public readonly elementType = InitElementType.COMMAND;
+
+  protected constructor(private readonly command: any, private readonly options: InitCommandOptions) {
+    super();
+
+    if (typeof this.command !== 'string' && !(Array.isArray(command) && command.every(s => typeof s === 'string'))) {
+      throw new Error('\'command\' must be either a string or an array of strings');
+    }
+  }
+
+  public bind(options: InitBindOptions): InitElementConfig {
+    const commandKey = this.options.key || `${options.index}`.padStart(3, '0'); // 001, 005, etc.
+
+    if (options.platform !== InitPlatform.WINDOWS && this.options.waitAfterCompletion !== undefined) {
+      throw new Error(`Command '${this.command}': 'waitAfterCompletion' is only valid for Windows systems.`);
+    }
+
+    for (const handle of this.options.serviceRestartHandles ?? []) {
+      handle.addCommand(commandKey);
+    }
+
+    return {
+      config: {
+        [commandKey]: {
+          command: this.command,
+          env: this.options.env,
+          cwd: this.options.cwd,
+          test: this.options.test,
+          ignoreErrors: this.options.ignoreErrors,
+          waitAfterCompletion: this.options.waitAfterCompletion?.toSeconds(),
+        },
+      },
+    };
+  }
+
+}
+
+/**
+ * Options for an InitService
+ */
+export interface InitServiceOptions {
+  /**
+   * Enable or disable this service
+   *
+   * Set to true to ensure that the service will be started automatically upon boot.
+   *
+   * Set to false to ensure that the service will not be started automatically upon boot.
+   *
+   * @default - true if used in `InitService.enable()`, no change to service
+   * state if used in `InitService.fromOptions()`.
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * Make sure this service is running or not running after cfn-init finishes.
+   *
+   * Set to true to ensure that the service is running after cfn-init finishes.
+   *
+   * Set to false to ensure that the service is not running after cfn-init finishes.
+   *
+   * @default - same value as `enabled`.
```

Comment:
repeat rather than asking the user to look it up.

Reply:
This actually means it defaults to the same value as `enabled`, not that it follows the same default logic as enabled. If you explicitly set enabled to false and don't set ensureRunning, it gets set to false.